File: app/src/main/java/com/productivity/habits/ui/daily/RollingWeekStrip.kt
package com.productivity.habits.ui.daily

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowLeft
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Today
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalHapticFeedback
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.productivity.habits.ui.common.HapticsHelper
import java.time.LocalDate
import java.time.format.DateTimeFormatter
import java.time.format.TextStyle
import java.util.Locale

@Composable
fun RollingWeekStrip(
    selectedDate: LocalDate,
    weekLogs: Map<LocalDate, Int>,
    onDateSelected: (LocalDate) -> Unit,
    onPreviousDay: () -> Unit,
    onNextDay: () -> Unit,
    onTodayClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val haptic = LocalHapticFeedback.current
    val isToday = selectedDate == LocalDate.now()

    // 7 days centered on selected date (-3 to +3)
    val days = (-3..3).map { selectedDate.plusDays(it.toLong()) }

    Surface(
        modifier = modifier.fillMaxWidth(),
        color = MaterialTheme.colorScheme.surface,
        tonalElevation = 1.dp
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(vertical = 8.dp, horizontal = 12.dp)
        ) {
            // Header Row: Steppers + Month/Year + Today button if not today
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    IconButton(
                        onClick = {
                            HapticsHelper.performLightHaptic(haptic)
                            onPreviousDay()
                        },
                        modifier = Modifier.size(36.dp)
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.KeyboardArrowLeft,
                            contentDescription = "Previous Day",
                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }

                    val monthFormatter = DateTimeFormatter.ofPattern("MMMM yyyy")
                    Text(
                        text = selectedDate.format(monthFormatter),
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.SemiBold,
                        color = MaterialTheme.colorScheme.onSurface,
                        modifier = Modifier.padding(horizontal = 4.dp)
                    )

                    IconButton(
                        onClick = {
                            HapticsHelper.performLightHaptic(haptic)
                            onNextDay()
                        },
                        modifier = Modifier.size(36.dp)
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.KeyboardArrowRight,
                            contentDescription = "Next Day",
                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }

                if (!isToday) {
                    TextButton(
                        onClick = {
                            HapticsHelper.performLightHaptic(haptic)
                            onTodayClick()
                        },
                        modifier = Modifier.height(32.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Today,
                            contentDescription = null,
                            modifier = Modifier.size(16.dp),
                            tint = MaterialTheme.colorScheme.primary
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text(
                            text = "Today",
                            style = MaterialTheme.typography.labelMedium,
                            fontWeight = FontWeight.SemiBold,
                            color = MaterialTheme.colorScheme.primary
                        )
                    }
                }
            }

            Spacer(modifier = Modifier.height(8.dp))

            // 7-day strip
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                days.forEach { date ->
                    val isSelected = date == selectedDate
                    val isDayToday = date == LocalDate.now()
                    val completedCount = weekLogs[date] ?: 0

                    DayStripItem(
                        date = date,
                        isSelected = isSelected,
                        isToday = isDayToday,
                        completedCount = completedCount,
                        onClick = {
                            HapticsHelper.performLightHaptic(haptic)
                            onDateSelected(date)
                        },
                        modifier = Modifier.weight(1f)
                    )
                }
            }
        }
    }
}

@Composable
private fun DayStripItem(
    date: LocalDate,
    isSelected: Boolean,
    isToday: Boolean,
    completedCount: Int,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    val dayName = date.dayOfWeek.getDisplayName(TextStyle.SHORT, Locale.getDefault()).take(3)
    val dayNum = date.dayOfMonth.toString()

    val backgroundColor = when {
        isSelected -> MaterialTheme.colorScheme.primary
        isToday -> MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.4f)
        else -> Color.Transparent
    }

    val contentColor = when {
        isSelected -> MaterialTheme.colorScheme.onPrimary
        isToday -> MaterialTheme.colorScheme.primary
        else -> MaterialTheme.colorScheme.onSurface
    }

    val dayNameColor = when {
        isSelected -> MaterialTheme.colorScheme.onPrimary.copy(alpha = 0.85f)
        isToday -> MaterialTheme.colorScheme.primary
        else -> MaterialTheme.colorScheme.onSurfaceVariant
    }

    Column(
        modifier = modifier
            .padding(horizontal = 2.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(backgroundColor)
            .clickable(onClick = onClick)
            .padding(vertical = 8.dp, horizontal = 4.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Text(
            text = dayName,
            style = MaterialTheme.typography.labelSmall,
            fontWeight = FontWeight.Medium,
            color = dayNameColor
        )

        Spacer(modifier = Modifier.height(4.dp))

        Text(
            text = dayNum,
            style = MaterialTheme.typography.bodyMedium,
            fontWeight = if (isSelected || isToday) FontWeight.Bold else FontWeight.Normal,
            color = contentColor
        )

        Spacer(modifier = Modifier.height(6.dp))

        // Indicator dot for completion
        Box(
            modifier = Modifier
                .size(6.dp)
                .clip(CircleShape)
                .background(
                    when {
                        completedCount > 0 && isSelected -> MaterialTheme.colorScheme.onPrimary
                        completedCount > 0 -> MaterialTheme.colorScheme.primary
                        else -> Color.Transparent
                    }
                )
        )
    }
}

File: lib/ui/daily/widgets/rolling_week_strip.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../../common/haptics_helper.dart';

class RollingWeekStrip extends StatelessWidget {
  final DateTime selectedDate;
  final Map<DateTime, int> weekLogs;
  final ValueChanged<DateTime> onDateSelected;
  final VoidCallback onPreviousDay;
  final VoidCallback onNextDay;
  final VoidCallback onTodayClick;

  const RollingWeekStrip({
    super.key,
    required this.selectedDate,
    required this.weekLogs,
    required this.onDateSelected,
    required this.onPreviousDay,
    required this.onNextDay,
    required this.onTodayClick,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final now = DateTime.now();
    final today = DateTime(now.year, now.month, now.day);
    final currentSelected = DateTime(
      selectedDate.year,
      selectedDate.month,
      selectedDate.day,
    );
    final isToday = currentSelected == today;

    final days = List.generate(7, (index) {
      final offset = index - 3; // -3 to +3
      final d = currentSelected.add(Duration(days: offset));
      return DateTime(d.year, d.month, d.day);
    });

    final monthFormatter = DateFormat('MMMM yyyy');

    return Material(
      color: theme.colorScheme.surface,
      elevation: 1,
      child: Padding(
        padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 12),
        child: Column(
          children: [
            // Header Row: Steppers + Month/Year + Today button
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: [
                Row(
                  children: [
                    IconButton(
                      iconSize: 20,
                      constraints: const BoxConstraints(
                        minWidth: 36,
                        minHeight: 36,
                      ),
                      padding: EdgeInsets.zero,
                      icon: Icon(
                        Icons.chevron_left,
                        color: theme.colorScheme.onSurfaceVariant,
                      ),
                      onPressed: () {
                        HapticsHelper.performLightHaptic();
                        onPreviousDay();
                      },
                    ),
                    Padding(
                      padding: const EdgeInsets.symmetric(horizontal: 4),
                      child: Text(
                        monthFormatter.format(selectedDate),
                        style: theme.textTheme.titleMedium?.copyWith(
                          fontWeight: FontWeight.w600,
                          color: theme.colorScheme.onSurface,
                        ),
                      ),
                    ),
                    IconButton(
                      iconSize: 20,
                      constraints: const BoxConstraints(
                        minWidth: 36,
                        minHeight: 36,
                      ),
                      padding: EdgeInsets.zero,
                      icon: Icon(
                        Icons.chevron_right,
                        color: theme.colorScheme.onSurfaceVariant,
                      ),
                      onPressed: () {
                        HapticsHelper.performLightHaptic();
                        onNextDay();
                      },
                    ),
                  ],
                ),
                if (!isToday)
                  TextButton.icon(
                    style: TextButton.styleFrom(
                      padding: const EdgeInsets.symmetric(
                        horizontal: 8,
                        vertical: 4,
                      ),
                      minimumSize: const Size(0, 32),
                    ),
                    icon: Icon(
                      Icons.today,
                      size: 16,
                      color: theme.colorScheme.primary,
                    ),
                    label: Text(
                      'Today',
                      style: theme.textTheme.labelMedium?.copyWith(
                        fontWeight: FontWeight.w600,
                        color: theme.colorScheme.primary,
                      ),
                    ),
                    onPressed: () {
                      HapticsHelper.performLightHaptic();
                      onTodayClick();
                    },
                  ),
              ],
            ),
            const SizedBox(height: 8),

            // 7-day strip
            Row(
              mainAxisAlignment: MainAxisAlignment.spaceBetween,
              children: days.map((date) {
                final isSelected = date == currentSelected;
                final isDayToday = date == today;
                final completedCount = weekLogs[date] ?? 0;

                return Expanded(
                  child: _DayStripItem(
                    date: date,
                    isSelected: isSelected,
                    isToday: isDayToday,
                    completedCount: completedCount,
                    onClick: () {
                      HapticsHelper.performLightHaptic();
                      onDateSelected(date);
                    },
                  ),
                );
              }).toList(),
            ),
          ],
        ),
      ),
    );
  }
}

class _DayStripItem extends StatelessWidget {
  final DateTime date;
  final bool isSelected;
  final bool isToday;
  final int completedCount;
  final VoidCallback onClick;

  const _DayStripItem({
    required this.date,
    required this.isSelected,
    required this.isToday,
    required this.completedCount,
    required this.onClick,
  });

  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    final dayName = DateFormat('E').format(date).substring(0, 3);
    final dayNum = date.day.toString();

    final Color backgroundColor;
    if (isSelected) {
      backgroundColor = theme.colorScheme.primary;
    } else if (isToday) {
      backgroundColor = theme.colorScheme.primaryContainer.withValues(alpha: 0.4);
    } else {
      backgroundColor = Colors.transparent;
    }

    final Color contentColor;
    if (isSelected) {
      contentColor = theme.colorScheme.onPrimary;
    } else if (isToday) {
      contentColor = theme.colorScheme.primary;
    } else {
      contentColor = theme.colorScheme.onSurface;
    }

    final Color dayNameColor;
    if (isSelected) {
      dayNameColor = theme.colorScheme.onPrimary.withValues(alpha: 0.85);
    } else if (isToday) {
      dayNameColor = theme.colorScheme.primary;
    } else {
      dayNameColor = theme.colorScheme.onSurfaceVariant;
    }

    final Color dotColor;
    if (completedCount > 0 && isSelected) {
      dotColor = theme.colorScheme.onPrimary;
    } else if (completedCount > 0) {
      dotColor = theme.colorScheme.primary;
    } else {
      dotColor = Colors.transparent;
    }

    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 2),
      child: Material(
        color: backgroundColor,
        borderRadius: BorderRadius.circular(12),
        child: InkWell(
          borderRadius: BorderRadius.circular(12),
          onTap: onClick,
          child: Padding(
            padding: const EdgeInsets.symmetric(vertical: 8, horizontal: 4),
            child: Column(
              mainAxisSize: MainAxisSize.min,
              children: [
                Text(
                  dayName,
                  style: theme.textTheme.labelSmall?.copyWith(
                    fontWeight: FontWeight.w500,
                    color: dayNameColor,
                  ),
                ),
                const SizedBox(height: 4),
                Text(
                  dayNum,
                  style: theme.textTheme.bodyMedium?.copyWith(
                    fontWeight:
                        (isSelected || isToday) ? FontWeight.bold : FontWeight.normal,
                    color: contentColor,
                  ),
                ),
                const SizedBox(height: 6),
                Container(
                  width: 6,
                  height: 6,
                  decoration: BoxDecoration(
                    color: dotColor,
                    shape: BoxShape.circle,
                  ),
                ),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
